File: 26.go
package main

func removeDuplicates(nums []int) int {
    unique := 0

    for i := 0; i < len(nums); i++ {

        if nums[i] != nums[unique] {
            unique++
            nums[unique] = nums[i]
        }
    }

    unique++

    return unique
}

func main() {
    //removeDuplicates([]int{0,0,1,1,1,2,2,3,3,4})
    removeDuplicates([]int{1, 1, 2})
}

File: 26.py
from typing import List


def removeDuplicates(nums: List[int]) -> int:
    i, j = 0, 1

    while j < len(nums):
        if nums[i] == nums[j]:
            j += 1
            continue

        i += 1
        nums[i] = nums[j]
        j += 1

    return i + 1


nums = [1, 1, 2]
nums2 = [0, 0, 1, 1, 1, 2, 2, 3, 3, 4]
print(removeDuplicates(nums))
print(removeDuplicates(nums2))
